src/api_recievers/influxqueries.js

```diff
@@ -0,0 +1,204 @@
+// const { time } = require("console");
+// const { start } = require("repl");
+
+const { currentTime, InfluxDB, Point } = require("@influxdata/influxdb-client");
+
+const log = reuqire('../util/logger');
+
+
+const token = process.env.INFLUXDB_TOKEN;
+const url = 'http://localhost:8086';
+const client = new InfluxDB({url, token});
+let org = `Distraction`;
+let bucket = `WebsiteData`;
+
+
+//Write Function
+//Source is from Chrome API or OS 
+//Must provide the application_name
+//Must provide the current study session this entry is associaed with
+//Inputs are Strings
+const appData = (source, appName, currentSession) =>{
+  let writeClient = client.getWriteApi(org, bucket, 's');
+  
+  let app = new Point('AppChange').stringField('AppName',appName).timestamp(currentTime.seconds()).stringField('Source', source).tag('QuerySession',currentSession);
+  writeClient.writePoint(app);
+  log.debug(`App added to InfluxDB: 
+    Name: ${appName}, 
+    Source: ${source}, 
+    QuerySession: ${currentSession}, 
+    Timestamp: ${currentTime.seconds()}`);
+  //console.log(currentTime.seconds())
+  writeClient.close();
+
+};
+
+
+// Function that will extract the time and the query session associated with that time 
+// Can add in the start and the end time of the measurement 
+const grabTimesForStudySession = (querySessionID) => {
+  return new Promise((resolve, reject) => {
+    const allTimes = []; // exact time of point
+    const allqueryIds = []; // Associated Query Session ID of that point
+    const allStartTimes = []; // When the measurement was started
+    const allEndTimes = []; // When the measreuemnt was ended 
+
+
+    // Get the current time
+    const timeOfCurrentSession = currentTime.seconds();
+    let queryClient = client.getQueryApi(org);
+    let stringVersion = `-${timeOfCurrentSession.toString()}s`;
+
+    // Construct the Flux query with the filter for the specific querySessionID
+    const fluxQuery = ` from(bucket: "WebsiteData") |> range(start: ${stringVersion}) |> filter(fn: (r) => r._measurement == "AppChange" and r.QuerySession == "${querySessionID}")
+    `;
+
+    queryClient.queryRows(fluxQuery, {
+      next: (row, tableMeta) => {
+        const tableObject = tableMeta.toObject(row);
+        allTimes.push(tableObject._time);
+        allqueryIds.push(tableObject.QuerySession);
+        allStartTimes.push(tableObject._start);
+        allEndTimes.push(tableObject._stop);
+        
+      },
+      error: (error) => {
+        reject(error);
+      },
+      complete: () => {
+        resolve({ allTimes, allqueryIds, allStartTimes, allEndTimes });
+      }
+    });
+  });
+};
```

Comment:
We should change this to an async function for consistancy:
```
const grabTimesForStudySession = async (querySessionID) => {
    const allTimes = []; // exact time of point
    ...
    
    queryClient.queryRows(fluxQuery, {
      next: (row, tableMeta) => {
        const tableObject = tableMeta.toObject(row);
        allTimes.push(tableObject._time);
        allqueryIds.push(tableObject.QuerySession);
        allStartTimes.push(tableObject._start);
        allEndTimes.push(tableObject._stop);

      },
      error: (error) => {
        throw error;
      },
      complete: () => {
        return { allTimes, allqueryIds, allStartTimes, allEndTimes };
      }
    });
};
```
Im pretty sure this syntax should work, but definitely double check

Reply:
Done, will double check tmr.